### PR TITLE
Fix read provider security using direct DB access

### LINKED ISSUE #299: Permissions allow everybody to read a provider
Everybody can read a provider, please fix this security hole.

### PR DESCRIPTION
**Pull request by Emulebest**
resolves #299

### CODE REVIEW

File: papiea-engine/src/provider/provider_api_interface.ts [hunk 1]
```diff
@@ -31,6 +31,8 @@ export interface Provider_API {
 
     get_provider(user: UserAuthInfo, provider_prefix: string, provider_version: Version): Promise<Provider>;
 
+    get_provider_unchecked(provider_prefix: string, provider_version: Version): Promise<Provider>
```
Comment (shlomi-nx) [inline papiea-engine/src/provider/provider_api_interface.ts]: @Emulebest why is the unchecked version exposed through the APIs?

**Comment by shlomi-nx** [comment]
@Emulebest Are there any changes in APIs that this introduces? Anything needs to change in xi-cluster-provider?

**Comment by Emulebest** [comment]
> @Emulebest Are there any changes in APIs that this introduces? Anything needs to change in xi-cluster-provider?

I tested that locally, seems fine, need to run the tests on CI. Also this discussion belongs to a different project

**Comment by shlomi-nx** [comment]
@Emulebest Alright, assuming the answer to my previous question boils down to "no changes to the apis" you can merge it. If the answer is yes, you can still merge but open a ticket in the relevant project to have it fixed.

**Pull request merged**